src/lib/storage/pos_lists/entire_chunk_pos_list.hpp

```diff
@@ -7,10 +7,9 @@ namespace opossum {
 
 class EntireChunkPosList : public AbstractPosList {
  public:
-  explicit EntireChunkPosList(std::shared_ptr<const Chunk> common_chunk, const ChunkID common_chunk_id)
-      : _common_chunk(common_chunk),
+  explicit EntireChunkPosList(const ChunkID common_chunk_id, const ChunkOffset common_chunk_size) :
         _common_chunk_id(common_chunk_id),
-        _common_chunk_size_on_creation(common_chunk->size()) {}
+        _common_chunk_size(common_chunk_size) {}
```

Comment:
Could you add documentation that explains that `common_chunk_size` has to be determined before iterating over the segment? In Validate (which seems to be the only user right now), this is not important because the chunk has to be finalized - still, it should be documented there as well.

Reply:
Ah ok that explains what I asked in https://github.com/hyrise/hyrise/pull/2025#discussion_r399819697

Will do.

Reply:
Added:

> The EntireChunkPosList will match the first common_chunk_size elements in the chunk.
If you create an EntireChunkPosList for a mutable chunk, make sure to determine that size
_before_ doing the checks that all elements should match. This way, you prevent the race
condition of an element being added in between doing the checks and getting the size.

Reply:
top